src/components/BeverageForm.js

```diff
@@ -0,0 +1,229 @@
+// @flow
+
+import type {
+  Availability,
+  Beverage,
+  Glass,
+  Srm,
+  Style,
+} from 'brewskey.js-api';
+import type DAOEntityStore from '../stores/DAOEntityStore';
+import type { FormChildProps, FormFieldChildProps } from '../common/form/types';
+
+import * as React from 'react';
+import { inject, observer } from 'mobx-react';
+import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
+import { Button, FormValidationMessage } from 'react-native-elements';
+import Form from '../common/form/Form';
+import FormField from '../common/form/FormField';
+
+import TextField from './TextField';
+import Picker from './Picker';
+
+const YEARS_RANGE_LENGTH = 10;
+
+type Props = {
+  availabilityStore: DAOEntityStore<Availability, Availability>,
+  beverage?: Beverage,
+  glassStore: DAOEntityStore<Glass, Glass>,
+  onSubmit: (values: Beverage) => Promise<void>,
+  srmStore: DAOEntityStore<Srm, Srm>,
+  styleStore: DAOEntityStore<Style, Style>,
+  submitButtonLabel: string,
+};
+
+@inject('availabilityStore')
+@inject('glassStore')
+@inject('srmStore')
+@inject('styleStore')
+@observer
+class BeverageForm extends React.Component<Props> {
+  static defaultProps = {
+    beverage: {},
+  };
+
+  componentWillMount() {
+    // todo replace to fetchAll when it will be implemented
+    this.props.availabilityStore.fetchMany();
+    this.props.srmStore.fetchMany({
+      orderBy: [{ column: 'hex', direction: 'desc ' }],
+    });
+    this.props.glassStore.fetchMany();
+    this.props.styleStore.fetchMany();
+  }
+
+  // todo Implement custom component for srm
```

Comment:
I was going to say that a dropdown would work I'm guessing that's the native picker :/

Reply:
It works, I made the field with picker.. but I saw that in v1 you have custom slider